### PR TITLE
Add logging when registering a dataset

### PR DESCRIPTION
**Pull request by sandalns**
### Linked issue(s):
Fixes KOL-4158

### What change does this PR introduce and why?
Adds logging when registering a dataset (logs the url to the dataset as well), also adds some logging when fetching a dataset

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: kolena/_experimental/dataset/_dataset.py [hunk 1]
```diff
@@ -251,9 +253,12 @@ def register_dataset(
                 validate_dataframe_ids(chunk, id_fields)
                 validated = True
             _upload_dataset_chunk(chunk, load_uuid, id_fields)
+
     request = RegisterRequest(name=name, id_fields=id_fields, uuid=load_uuid)
     response = krequests.post(Path.REGISTER, json=asdict(request))
     krequests.raise_for_status(response)
+    data = from_dict(EntityData, response.json())
+    log.info(f"Successfully created/updated dataset {name} {get_dataset_url(dataset_id=data.id)}")
```
Comment (ly-kolena) [inline kolena/_experimental/dataset/_dataset.py]: Let's keep the link off until the feature is deployed to production.

**Review by ly-kolena — APPROVED** [review]

File: kolena/_experimental/dataset/_dataset.py (hunk 1 above)
Comment (gordonhart) [reply kolena/_experimental/dataset/_dataset.py]: Why?

Comment (gordonhart) [inline kolena/_experimental/dataset/_dataset.py]: In existing messages, we're careful to always surround names with single quotes (`'{name}'`) and URLs in parentheses (`({url})`). Let's follow that pattern here too.

Comment (gordonhart) [reply kolena/_experimental/dataset/_dataset.py]: Also, can we think of a better verb than `created/updated`? The message here should mirror the verb in the function name.

Comment (ly-kolena) [reply kolena/_experimental/dataset/_dataset.py]: It's not workable now, right?

Comment (gordonhart) [reply kolena/_experimental/dataset/_dataset.py]: True, but it doesn't feel any more or less broken to me to print success messages with a URL that redirects you away than to print success messages about content you have no way of reaching. In the interest of not forgetting to add URLs in a follow-up prior to release, I'm fine with adding links before this goes live in January.

Comment (ly-kolena) [reply kolena/_experimental/dataset/_dataset.py]: Ack

File: kolena/_experimental/dataset/_dataset.py [hunk 2]
```diff
@@ -292,4 +297,5 @@ def fetch_dataset(
     Fetch an entire dataset given its name.
     """
     df_batches = list(iter_dataset(name, batch_size))
+    log.info(f"Loaded dataset {name}")
```
Comment (gordonhart) [inline kolena/_experimental/dataset/_dataset.py]: Same formatting comment here, and in all other messages updated in this PR.

File: kolena/_experimental/dataset/_dataset.py (hunk 1 above)
Comment (sandalns) [reply kolena/_experimental/dataset/_dataset.py]: Updated to use `registered` as it fits both creating and updating the dataset. Also added the single quotes and parentheses

File: kolena/_experimental/dataset/_evaluation.py [hunk 3]
```diff
@@ -128,7 +129,7 @@ def fetch_results(
                 _to_deserialized_dataframe(df_matched, column=COL_RESULT),
             ),
         )
-
+    log.info(f"Fetched results for dataset '{dataset}' and model '{model}'")
```
Comment (gordonhart) [inline kolena/_experimental/dataset/_evaluation.py]: Nit: let's use consistent wording between this function and `upload_results`:

> Fetched results for model 'name' on dataset 'name'

File: kolena/_experimental/dataset/_evaluation.py [hunk 4]
```diff
@@ -191,4 +192,4 @@ def test(
                 upload_data_frame(df=df_results, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
 
     _upload_results(model, load_uuid, existing_dataset.id)
-    log.info(f"uploaded test results for model {model} on dataset {dataset}")
+    log.info(f"uploaded test results for model '{model}' on dataset '{dataset}'")
```
Comment (gordonhart) [inline kolena/_experimental/dataset/_evaluation.py]: Nit: uppercase start of sentence

**Review by gordonhart — APPROVED** [review]

**Pull request merged**